### PR TITLE
Stats: Fix routes checking for Overview, Site, Summary, and WordAds

### PR DESCRIPTION
**Pull request by dognose24**
#### Proposed Changes

* ~Restore the `pathname` to compare paths without query string [as previous logic](https://github.com/Automattic/wp-calypso/blob/2436d081fb82fd0b18e25b0c5108033adc6e934d/client/my-sites/stats/controller.jsx#L191-L192).~
* Deprecate `altPaths` since there have been redirections handling the alternative paths.
* Extract `getWordAdsFilters` for the `Ads` page routes since `WordAds` [has its own routing](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L73).

#### Testing Instructions

> This change also aims to make routes work in the [Stats Standalone App for Jetpack plugins](https://github.com/Automattic/wp-calypso/pull/69231).

* Spin up this change with the Calypso Live link.
* Ensure the following paths work.
* Overview routes:
    - `/stats` ( [redirected](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L43) )
    - `/stats/day`
    - `/stats/week`
    - `/stats/month`
    - `/stats/year`
* Site routes:
    - `/stats/insights/${your-site}`
    - `/stats/${your-site}` ( [redirected](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L80) )
    - `/stats/day/${your-site}`
    - `/stats/week/${your-site}`
    - `/stats/month/${your-site}`
    - `/stats/year/${your-site}`
* WordAds routes:
    - `/stats/ads/${your-site}` ( [redirected](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L76-L77) )
    - `/stats/ads/day/${your-site}`
    - `/stats/ads/week/${your-site}`
    - `/stats/ads/month/${your-site}`
    - `/stats/ads/year/${your-site}`
* Summary routes (`posts` is one of the [valid modules](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L28-L39)):
    - `/stats/posts/${your-site}` ( [redirected](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L58) )
    - `/stats/day/posts/${your-site}`
    - `/stats/week/posts/${your-site}`
    - `/stats/month/posts/${your-site}`
    - `/stats/year/posts/${your-site}`

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70602
Related to #70563

### CODE REVIEW

**Review by kangzj — APPROVED** [review]
It works for me but probably better to get more people to have a go @Automattic/red

**Comment by dognose24** [comment]
> It works for me but probably better to get more people to have a go https://github.com/orgs/Automattic/teams/red

Yes, we need more eyes on the route checking for both Calypso Stats and the standalone app.

File: client/my-sites/stats/controller.jsx [hunk 1]
```diff
@@ -186,11 +194,7 @@ export function overview( context, next ) {
 	window.scrollTo( 0, 0 );
 
 	const activeFilter = find( filters(), ( filter ) => {
-		return (
-			context.params.period === filter.period ||
-			context.path.indexOf( filter.path ) >= 0 ||
-			( filter.altPaths && context.path.indexOf( filter.altPaths ) >= 0 )
-		);
+		return context.params.period === filter.period || context.path.indexOf( filter.path ) >= 0;
```
Comment (jsnmoon) [inline client/my-sites/stats/controller.jsx]: We could go further with this refactor and use [`String.prototype.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) like so:

```suggestion
		return context.params.period === filter.period || context.path.includes( filter.path );
```

File: client/my-sites/stats/controller.jsx [hunk 2]
```diff
@@ -219,10 +223,7 @@ export function site( context, next ) {
 	const siteId = currentSite ? currentSite.ID || 0 : 0;
 
 	const activeFilter = find( filters, ( filter ) => {
-		return (
-			context.path.indexOf( filter.path ) >= 0 ||
-			( filter.altPaths && context.path.indexOf( filter.altPaths ) >= 0 )
-		);
+		return context.path.indexOf( filter.path ) >= 0;
```
Comment (jsnmoon) [inline client/my-sites/stats/controller.jsx]:
```suggestion
		return context.path.includes( filter.path );
```

File: client/my-sites/stats/controller.jsx [hunk 3]
```diff
@@ -296,10 +296,7 @@ export function summary( context, next ) {
 	siteId = selectedSite ? selectedSite.ID || 0 : 0;
 
 	const activeFilter = find( filters, ( filter ) => {
-		return (
-			context.path.indexOf( filter.path ) >= 0 ||
-			( filter.altPaths && context.path.indexOf( filter.altPaths ) >= 0 )
-		);
+		return context.path.indexOf( filter.path ) >= 0;
```
Comment (jsnmoon) [inline client/my-sites/stats/controller.jsx]:
```suggestion
		return context.path.includes( filter.path );
```

**Review by jsnmoon — APPROVED** [review]
Works well for Calypso Stats. I did not test Odyssey Stats during my review.

**Comment by jsnmoon** [comment]
@dognose24: This PR appears to break [the "Click on Insights tab" test](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/stats/stats__insights.ts). Can you check?

```
locator.innerText: Timeout 10000ms exceeded.
=========================== logs ===========================
waiting for locator('div.section-nav a[aria-current="true"]:has(span:has-text(""))')
```

**Comment by dognose24** [comment]
> This PR appears to break [the "Click on Insights tab" test](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/stats/stats__insights.ts). Can you check?

@jsnmoon I think it is good now!

**Review by jsnmoon — APPROVED** [review]

**Pull request merged**